internal/curator/durable/fsm.go

```diff
@@ -187,10 +185,10 @@ func (cmd SyncPartitionsCommand) apply(txn *state.Txn) SyncPartitionsResult {
 // Creates a new blob in a partition that has space for it and returns it.
 func (cmd CreateBlobCommand) apply(txn *state.Txn) CreateBlobResult {
```

Comment:
Just curious why we are returning a core.Error in for (cmd CreatTractCommand) apply(...) and are still returning CreateBlobResult here?

Reply:
Each command gets to have its own return type. Think of it like an in-process RPC: the ___Command types are the requests, the ___Result types are the replies. For many commands, there's no reason to return anything, so we just return a core.Error. The methods in handler.go cast things to the right type.

Where this gets a little messy is that there are a couple cases where Apply can return something other than the expected return type. One is during replay when the transaction has already been applied. That's harmless since it doesn't go through Propose. The annoying one is read-only mode, where Apply can return ErrReadOnlyMode (a core.Error value) without applying the command. So all the handlers have to check for core.Error even if it's not their result type.

It turns out read-only mode is kinda broken and doesn't fully solve the problem it was meant to solve, and we should just remove it and fix it some other way. Then the handlers can assume they're getting the expected type.

Reply:
This makes sense!